apps/studio/components/interfaces/Storage/VectorBuckets/VectorBucketDetails.tsx

```diff
@@ -0,0 +1,227 @@
+import { useParams } from 'common'
+import { PageLayout } from 'components/layouts/PageLayout/PageLayout'
+import {
+  ScaffoldContainer,
+  ScaffoldHeader,
+  ScaffoldSection,
+  ScaffoldSectionDescription,
+  ScaffoldSectionTitle,
+} from 'components/layouts/Scaffold'
+import { DocsButton } from 'components/ui/DocsButton'
+import { useVectorBucketsIndexesQuery } from 'data/storage/vector-buckets-indexes-query'
+import { Eye, MoreVertical, Search, Trash2 } from 'lucide-react'
+import Link from 'next/link'
+import { useState } from 'react'
+import {
+  Button,
+  Card,
+  CardContent,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from 'ui'
+import { Input } from 'ui-patterns/DataInputs/Input'
+import { TimestampInfo } from 'ui-patterns/TimestampInfo'
+import { BUCKET_TYPES } from '../Storage.constants'
+import { CreateVectorIndexSheet } from './CreateVectorIndexSheet'
+
+// Mock data for development - remove when backend is ready
+const MOCK_VECTOR_INDEXES = [
+  {
+    indexName: 'documents_index',
+    creationTime: '1704067200', // Jan 1, 2024
+  },
+  {
+    indexName: 'embeddings_index',
+    creationTime: '1704153600', // Jan 2, 2024
+  },
+  {
+    indexName: 'search_index',
+    creationTime: '1704240000', // Jan 3, 2024
+  },
+]
+
+const MOCK_VECTOR_INDEXES_EMPTY = []
+
+interface VectorBucketDetailsProps {
+  bucket: { vectorBucketName: string; creationTime: string }
+}
+
+export const VectorBucketDetails = ({ bucket }: VectorBucketDetailsProps) => {
+  const { ref: projectRef } = useParams()
+
+  // Use the correct query for bucket contents
+  const { data, isLoading, error } = useVectorBucketsIndexesQuery({
+    projectRef,
+    vectorBucketName: bucket.vectorBucketName,
+  })
+
+  // Mock data for development - replace with real data when backend is ready
+  // const allIndexes = MOCK_VECTOR_INDEXES
```

Comment:
Use this instead of `data?.vectorBuckets` if you aren’t able to populate with real data.